aggregator/src/aggregator.rs

```diff
@@ -2173,6 +2186,39 @@ impl VdafOps {
                         if aggregation_job.last_request_hash()
                             != existing_aggregation_job.last_request_hash()
                         {
+                            if log_forbidden_mutations {
+                                let span = debug_span!(
+                                    parent: Span::current(),
+                                    "aggregation_job_init_forbidden_mutation_span",
+                                );
+                                let _span_entered = span.enter();
+                                let existing_report_metadatas: Vec<_> = tx
+                                    .get_report_aggregations_for_aggregation_job(
+                                        vdaf.as_ref(),
+                                        &Role::Helper,
+                                        task.id(),
+                                        aggregation_job.id(),
+                                    )
```

Comment:
I'm a little concerned that the order in which we'll get `report_aggregations` rows from this query won't be guaranteed to match the order in which they appeared in an `AggregationJobInitReq`, but I think they _should_.

Reply:
`get_report_aggregations_for_aggregation_job` retrieves report aggregations ordered by `ord`, which is the correct (required) order for the `AggregationJobInitReq`'s `prepare_inits`.